Source/JavaScriptCore/b3/B3ReduceStrength.cpp

```diff
@@ -1982,8 +1982,10 @@ class ReduceStrength {
             // Turn this: Trunc(constant)
             // Into this: static_cast<int32_t>(constant)
             if (m_value->child(0)->hasInt64() || m_value->child(0)->hasDouble()) {
-                replaceWithNewValue(
-                    m_proc.addIntConstant(m_value, static_cast<int32_t>(m_value->child(0)->asInt64())));
+                int32_t truncValue = m_value->child(0)->hasInt64()
```

Comment:
This looks not correct from how `Trunc(Double)` works from b3/intermediate-representation.html. See 2ea19e682e31953ec3116b53140f146a7458106c